### PR TITLE
Add Dart style guide

### PR DESCRIPTION
**Pull request by kwalrath**
* Added Go while I was in there. (@rakyll or @adg might have feedback on the link I used)
* Rewrote things a bit to make the new links fit.
* Copy edited a bit because I couldn't help myself.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,3 +49,6 @@ for more details.
 [cpplint]: https://github.com/google/styleguide/tree/gh-pages/cpplint
 [emacs]: https://raw.githubusercontent.com/google/styleguide/gh-pages/google-c-style.el
 [xml]: https://google.github.io/styleguide/xmlstyle.html
+[dart]: https://www.dartlang.org/guides/language/effective-dart
+[go]: https://github.com/golang/go/wiki/Style
```
Comment (rakyll) [inline README.md]: This is https://golang.org/wiki/CodeReviewComments now.

Comment (kwalrath) [reply README.md]: Yeah, I saw that. It's just a weirdly named page for a style guide, and I saw that Style linked to it. :) I'll update the link, though.

Comment (rakyll) [reply README.md]: Yeah, it is a bit weirdly named :) You can use the golang.org shortcut for the link btw: https://golang.org/wiki/CodeReviewComments. If we ever move the wiki somewhere, the link would still work.

Comment (kwalrath) [reply README.md]: SGTM. I'll do that now. Thanks for the feedback!

Comment (markmentovai) [reply README.md]: It looks like this list of links is ordered according to the sequence that things appear in the prose, but you inserted [go] before [dart] up there, and after it here.

**Comment by kwalrath** [comment]
Thanks!

**Pull request merged**